it/uk/gov/hmrc/agentsubscription/controllers/SubscriptionControllerISpec.scala

```diff
@@ -293,7 +293,166 @@ class SubscriptionControllerISpec extends BaseISpec with DesStubs with AuthStub
+      requestIsAuthenticated().andIsAnAgent().andHasNoEnrolments()
+      agentRecordExists(utr, true, arn)
+      allocatedPrincipalEnrolmentExists(arn, "someGroupId")
+
+      val result = await(doUpdateSubscriptionRequest())
+
+      result.status shouldBe 409
+
+      verify(0, deleteRequestedFor(urlEqualTo(s"$deleteKnownFactsUrl$arn")))
+      verify(0, putRequestedFor(urlEqualTo(s"$createKnownFactsUrl$arn")))
+      verify(0, postRequestedFor(urlEqualTo(enrolmentUrl("groupId", arn))))
+    }
+
+    "return forbidden" when {
+      "no registration exists" in {
+        requestIsAuthenticated().andIsAnAgent().andHasNoEnrolments()
+        agentRecordDoesNotExist(utr)
+
+        val result = await(doUpdateSubscriptionRequest())
+
+        result.status shouldBe 403
+      }
+
+      "postcodes don't match" in {
+        requestIsAuthenticated().andIsAnAgent().andHasNoEnrolments()
+        agentRecordDoesNotExist(utr)
+        val request = Json.parse(updateSubscriptionRequest).as[UpdateSubscriptionRequest].copy(knownFacts = KnownFacts("AA1 2AA"))
+
+        val result = await(doUpdateSubscriptionRequest(stringify(toJson(request))))
+
+        result.status shouldBe 403
+      }
+
+      "the user already has enrolments" in {
+        requestIsAuthenticated().andIsAnAgent().andHasEnrolments()
+        val result = await(doUpdateSubscriptionRequest())
+
+        result.status shouldBe 403
+      }
+    }
+
+    "return Bad Request " when {
+      "utr is missing" in {
+        requestIsAuthenticated()
+        val result = await(doUpdateSubscriptionRequest(removeFields(Seq(__ \ "utr"))))
+
+        result.status shouldBe 400
+      }
+
+      "utr contains non-numeric characters" in {
+        requestIsAuthenticated()
+        val result = await(doUpdateSubscriptionRequest(replaceFields(Seq((__, "utr", "ABCDE12345")))))
+
+        result.status shouldBe 400
+      }
+
+      "utr contains fewer than 10 digits" in {
+        requestIsAuthenticated()
+        val result = await(doUpdateSubscriptionRequest(replaceFields(Seq((__, "utr", "12345")))))
+
+        result.status shouldBe 400
+      }
+
+      "utr contains more than 10 digits" in {
+        requestIsAuthenticated()
+        val result = await(doUpdateSubscriptionRequest(replaceFields(Seq((__, "utr", "12345678901")))))
+
+        result.status shouldBe 400
+      }
+
+      "postcode is missing" in {
+        requestIsAuthenticated()
+        val result = await(doUpdateSubscriptionRequest(replaceFields(Seq((__ \ "knownFacts", "postcode", "")))))
+
+        result.status shouldBe 400
+      }
+
+      "known facts postcode is not valid" in {
+        requestIsAuthenticated()
+        val result = await(doUpdateSubscriptionRequest(replaceFields(Seq((__ \ "knownFacts", "postcode", "1234567")))))
+
+        result.status shouldBe 400
+      }
+    }
+
+    "throw a 500 error if " when {
```

Comment:
organise these tests, or name the test appropriately. Make it clear this is partiallySubscribed fix.